packages/hash/frontend/src/blocks/page/Comments/CommentTextField.tsx

```diff
@@ -0,0 +1,194 @@
+import { FunctionComponent, useCallback, useRef, useState } from "react";
+import { EditorState } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+import { Schema } from "prosemirror-model";
+import { formatKeymap } from "@hashintel/hash-shared/createProseMirrorState";
+import { baseKeymap } from "prosemirror-commands";
+import { keymap } from "prosemirror-keymap";
+import {
+  createSchema,
+  hardBreakNode,
+  mentionNode,
+  textNode,
+} from "@hashintel/hash-shared/prosemirror";
+import { Box } from "@mui/material";
+import { faAt } from "@fortawesome/free-solid-svg-icons";
+import {
+  IconButton,
+  FontAwesomeIcon,
+  LoadingSpinner,
+} from "@hashintel/hash-design-system";
+import { faComment } from "@fortawesome/free-regular-svg-icons";
+import { TextToken } from "@hashintel/hash-shared/graphql/types";
+import { usePortals } from "../usePortals";
+import { createFormatPlugins } from "../createFormatPlugins";
+import {
+  createSuggester,
+  suggesterPluginKey,
+} from "../createSuggester/createSuggester";
+import { useRouteAccountInfo } from "../../../shared/routing";
+import styles from "./style.module.css";
+import { placeholderPlugin } from "./placeholderPlugin";
+import { clipboardTextSerializer } from "../clipboardTextSerializer";
+import { mentionNodeView } from "../MentionView/MentionNodeView";
+
+type CommentTextFieldProps = {
+  blockId: string;
+  onClose: () => void;
+  onSubmit: ((content: TextToken[]) => Promise<void>) | null;
+};
+
+export const CommentTextField: FunctionComponent<CommentTextFieldProps> = ({
+  blockId,
+  onClose,
+  onSubmit,
+}) => {
+  const viewRef = useRef<EditorView<Schema>>();
+  const [portals, renderPortal] = usePortals();
+  const { accountId } = useRouteAccountInfo();
+  const [loading, setLoading] = useState(false);
+
+  const [prevBlockId, setPrevBlockId] = useState("");
+  const [editorContainer, setEditorContainer] = useState<HTMLDivElement | null>(
+    null,
+  );
+
+  const createEditor = useCallback(
+    (container: HTMLDivElement) => {
+      const schema = createSchema({
+        doc: {
+          content: "inline*",
+        },
+        text: textNode,
+        hardBreak: hardBreakNode,
+        mention: mentionNode,
+      });
+
+      const doc = schema.node("doc", {}, []);
+
+      const state = EditorState.create<Schema>({
+        doc,
+        plugins: [
+          keymap<Schema>(baseKeymap),
+          ...createFormatPlugins(renderPortal),
+          formatKeymap(doc),
+          createSuggester(renderPortal, accountId, container),
+          placeholderPlugin(renderPortal, "Leave a comment"),
+        ],
```

Comment:
Not sure if I see a lot of value in that, since the only plugins I would be able to include would be `...createFormatPlugins(renderPortal)`, and `createSuggester(renderPortal, accountId, container)`. The only other plugin that is also shared by both is `formatKeymap` but it is also required in the collab `createProsemirrorState` calls.